### DISCUSSION TITLE
Using BuildKit and Platform?

### DISCUSSION THREAD

**Discussion by Alexhuszagh** [lima-vm/lima#1075] [Q&A] [answered]
I'm currently attempting to build an image using `nerdctl.lima` on a Linux host with the `--platform` key. The build invocation looks something like:

```bash
$ nerdctl.lima buildx build --platform linux/amd64 -f Dockerfile .
FATA[0000] unknown flag: --platform
```

I'm currently using nerdctl version 0.23.0 and ctr containerd.io 1.6.8. I've also manually installed buildkit using `brew install buildkit`. I do not have `buildkitd` on my system, but `buildctl` exists (I'm assuming `buildkitd` is macOS specific, as the [quick start](https://github.com/moby/buildkit#quick-start) seems to suggest). Any help would be useful.

Skipping `buildx` and using `build` directly works, however:

```bash
$ nerdctl.lima build --platform linux/amd64 -f Dockerfile .
[+] Building 0.7s (2/3)                                                                                                                                                                             
 => [internal] load .dockerignore                                                                                                                                                              0.3s
 => => transferring context: 2B
...
```

**Comment by AkihiroSuda** [comment, marked as answer]
Expected behavior. `nerdctl buildx` does not exist.

`nerdctl build` supports multi-platform builds.

**Reply by Alexhuszagh** [reply]
Thanks for the quick reply. Would there anyway to improve the diagnostics then for subcommands that are specifically supported by Docker but not provided by `nerdctl`? I understand you're limited in the general case since the flags take priority over the subcommand for everything else (as is consistent with Docker and Podman).

I'm not sure of the exact API limitations of this, so this might be next to impossible even for this specific case.
